src/views/dialogs/DepositDialog2/queries.ts

```diff
@@ -109,3 +112,37 @@ function networkTypeToBalances(sourceAccount: SourceAccount): BalanceRequest {
 
   throw new Error('Fetching balances for unknown chain');
 }
+
+async function getSkipRoutes(skipClient: SkipClient, token: DepositToken, amount: string) {
+  const routeOptions: RouteRequest = {
+    sourceAssetDenom: token.denom,
+    sourceAssetChainID: token.chainId,
+    destAssetDenom: DYDX_CHAIN_USDC_DENOM,
+    destAssetChainID: DYDX_DEPOSIT_CHAIN,
+    amountIn: parseUnits(amount, token.decimals).toString(),
+    smartRelay: true,
+    smartSwapOptions: { evmSwaps: true },
+  };
+
+  const [slow, fast] = await Promise.all([
+    skipClient.route(routeOptions),
+    skipClient.route({ ...routeOptions, goFast: true }),
+  ]);
+
+  // @ts-ignore SDK doesn't know about .goFastTransfer
+  const isFastRouteAvailable = Boolean(fast.operations.find((op) => op.goFastTransfer));
+  return { slow, fast: isFastRouteAvailable ? fast : undefined };
+}
+
+export function useRoutes(token: DepositToken, amount: string) {
+  const { skipClient } = useSkipClient();
+  const rawAmount = amount && parseUnits(amount, token.decimals);
+  return useQuery({
+    queryKey: ['routes', token.chainId, token.denom, amount],
+    queryFn: () => getSkipRoutes(skipClient, token, amount),
+    enabled: Boolean(rawAmount && rawAmount > 0),
+    staleTime: 1 * timeUnits.minute,
+    refetchOnMount: 'always',
+    placeholderData: (prev) => prev,
```

Comment:
this feature is pretty neat - it basically retains the data from the previous query if the query is currently being fired with new parameters. useful for preventing UI trash if current data shows stuff on page but new query would otherwise cause stuff on page to disappear